cmd/server-mux.go

```diff
@@ -70,37 +65,35 @@ func longestWord(strings []string) int {
 			maxLen = len(m)
 		}
 	}
-
 	return maxLen
 }
 
 // NewConnMux - creates a new ConnMux instance
 func NewConnMux(c net.Conn) *ConnMux {
-	h1 := longestWord(defaultHTTP1Methods)
-	h2 := longestWord(defaultHTTP2Methods)
-	max := h1
-	if h2 > max {
-		max = h2
+	br := bufio.NewReader(c)
+	bw := bufio.NewWriter(c)
+	return &ConnMux{
+		Conn:    c,
+		bufwr:   bufio.NewReadWriter(br, bw),
+		readLen: longestWord() + 1,
 	}
-	return &ConnMux{Conn: c, dataBuf: ConnBuf{buffer: make([]byte, max+1)}}
 }
 
 // PeekProtocol - reads the first bytes, then checks if it is similar
 // to one of the default http methods
 func (c *ConnMux) PeekProtocol() string {
-	var n int
-	n, c.lastError = c.Conn.Read(c.dataBuf.buffer)
-	if n == 0 || (c.lastError != nil && c.lastError != io.EOF) {
-		return ""
+	var buf []byte
+	buf, c.lastError = c.bufwr.Peek(c.readLen)
```

Comment:
since c.readLen will always be the same value, it can be inited to a global constant and removed from ConnMux